src/Sentry.Maui/Internal/MauiDeviceData.cs

```diff
@@ -6,29 +6,22 @@ namespace Sentry.Maui.Internal;
 
 internal static class MauiDeviceData
 {
-    public static void ApplyMauiDeviceData(this Device device, IDiagnosticLogger? logger)
+    public static void ApplyMauiDeviceData(this Device device, IDiagnosticLogger? logger, INetworkStatusListener? networkStatusListener, string deviceName, string deviceManufacturer, string deviceModel, string deviceType, bool? deviceSimulator, DevicePlatform devicePlatform, DisplayInfo displayInfo, bool supportsVibration, bool supportsAccelerometer, bool supportsGyroscope)
     {
         try
         {
             // TODO: Add more device data where indicated
 
-            // https://docs.microsoft.com/dotnet/maui/platform-integration/device/information
-            var deviceInfo = DeviceInfo.Current;
-            if (deviceInfo.Platform == DevicePlatform.Unknown)
+            if (devicePlatform == DevicePlatform.Unknown)
             {
                 // return early so we don't get NotImplementedExceptions (i.e., in unit tests, etc.)
                 return;
             }
-            device.Name ??= deviceInfo.Name;
-            device.Manufacturer ??= deviceInfo.Manufacturer;
-            device.Model ??= deviceInfo.Model;
-            device.DeviceType ??= deviceInfo.Idiom.ToString();
-            device.Simulator ??= deviceInfo.DeviceType switch
-            {
-                DeviceType.Virtual => true,
-                DeviceType.Physical => false,
-                _ => null
-            };
+            device.Name ??= deviceName;
```

Comment:
Passing a `DeviceInfo` parameter to `ApplyMauiDeviceData()` and calling deviceInfo.Name causes the crash to happen on this very line.

I'm trying to find a solution to avoid a crash here, and avoid the [above problem](https://github.com/getsentry/sentry-dotnet/pull/3791/files#r1858416894)